modules/platforms/cpp/binary/include/ignite/impl/binary/binary_type_impl.h

```diff
@@ -56,85 +81,161 @@ namespace ignite
+            };
+
+            /**
+             * Specialization for the pointer case.
+             */
+            template<typename T>
+            struct WriteHelper<T*>
             {
-                if (!val)
-                    writer.WriteNull0();
-                else
-                    writer.template WriteTopObject0<BinaryWriter>(*val);
-            }
-        };
-
-        /**
-         * Read helper. Takes care of proper reading of pointers.
-         */
-        template<typename T>
-        struct ReadHelper
-        {
-            template<typename R>
-            static T Read(R& reader)
+                template<typename W>
+                static void Write(W& writer, const T* val)
+                {
+                    if (!val)
+                        writer.WriteNull0();
+                    else
+                        writer.template WriteTopObject0<ignite::binary::BinaryWriter>(*val);
+                }
+            };
+
+            /**
+             * Read helper. Takes care of proper reading of pointers.
+             */
+            template<typename T>
+            struct ReadHelper
             {
-                T res;
-
-                Read<R>(reader, res);
-
-                return res;
-            }
-
-            template<typename R>
-            static void Read(R& reader, T& val)
+                template<typename R>
+                static T Read(R& reader)
+                {
+                    T res;
+
+                    Read<R>(reader, res);
+
+                    return res;
+                }
+
+                template<typename R>
+                static void Read(R& reader, T& val)
+                {
+                    reader.template ReadTopObject0<ignite::binary::BinaryReader, T>(val);
+                }
+            };
+
+            /**
+             * Specialization for the pointer case.
+             */
+            template<typename T>
+            struct ReadHelper<T*>
             {
-                reader.template ReadTopObject0<BinaryReader, T>(val);
-            }
-        };
-
-        /**
-         * Specialization for the pointer case.
-         */
-        template<typename T>
-        struct ReadHelper<T*>
-        {
-            template<typename R>
-            static T* Read(R& reader)
+                template<typename R>
+                static T* Read(R& reader)
+                {
+                    if (reader.SkipIfNull())
+                        return 0;
+
+                    std::auto_ptr<T> res(new T());
+
+                    reader.template ReadTopObject0<ignite::binary::BinaryReader, T>(*res);
+
+                    return res.release();
+                }
+
+                template<typename R>
+                static void Read(R& reader, T*& ptr)
+                {
+                    ptr = Read<R>(reader);
+                }
+            };
+
+            IGNITE_DECLARE_BINARY_TYPE_METHOD_CHECKER(GetAffinityFieldName, void(*)(std::string&));
```

Comment:
My only concern is - if you make a typo in the method signature, the code will compile, but affinity won't work as expected.

Reply:
Well, it seems like there is no good solution for this - we ignore it if method is not present or breaking backward compatibility for our current users. I prefer the first approach.